lib/events/auditlog.go

```diff
@@ -1086,20 +1085,20 @@ func (l *AuditLog) StreamSessionEvents(ctx context.Context, sessionID string, st
 	}
 
 	protoReader := NewProtoReader(rawStream)
+	e := make(chan error)
 	c := make(chan apievents.AuditEvent)
-	ctx, cancel := context.WithCancel(ctx)
 
 	go func() {
 		for {
 			if ctx.Err() != nil {
-				close(c)
+				e <- trace.Wrap(ctx.Err())
 				break
 			}
 
 			event, err := protoReader.Read(ctx)
 			if err != nil {
 				if err != io.EOF {
-					cancel()
+					e <- trace.Wrap(err)
 				}
 
 				close(c)
```

Comment:
In the first error case you don't close `c`, in the second you do.

At first I thought the caller wouldn't be guaranteed to see the error before seeing that `c` has been closed, but these channels are unbuffered so the sends will be blocking and it should be fine. I still think it would be good to be more strict about which channels will be closed in which cases, though

Reply:
Adjusted to always close c on error.